backend/inspirehep/records/tasks.py

```diff
@@ -74,8 +74,22 @@ def update_records_relations(self, uuids):
 )
 def redirect_references_to_merged_record(self, uuid):
     record = InspireRecord.get_record(uuid, with_deleted=True)
-    new_record_ref = record["new_record"]["$ref"]
-    deleted_record_ref = record["self"]["$ref"]
+    new_record = record.get("new_record")
+    deleted_record = record.get("self")
+    if not new_record:
+        LOGGER.error(
+            "redirect_references_to_merged_record: Missing 'new_record' field for record with uuid.",
+            uuid=str(uuid),
+        )
+        return
+    if not deleted_record:
+        LOGGER.error(
+            "redirect_references_to_merged_record: Missing 'self' field for record with uuid.",
+            uuid=str(uuid),
+        )
+        return
```

Comment:
if both are missing it will never go to `deleted_record`, right?

Reply:
Updated to the first solution with more granular error message